### PR TITLE
Add `Proxy` to protect polyfill implementation.

### PR DESCRIPTION
**Pull request by dlongley**

### CODE REVIEW

**Review by BigBlueHat — APPROVED** [review]

**Comment by dlongley** [comment]
Addresses #38.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,12 @@
 # credential-handler-polyfill ChangeLog
 
+## 3.2.1 - 2024-05-dd
+
+### Fixed
+- Ensure `navigator.credentials[get|store]` methods from the polyfill
+  will continue to run to process `options.web|WebCredential` if other
+  software that runs after it is setup changes those methods.
```
Comment (TallTed) [inline CHANGELOG.md]:
```suggestion
  will continue to run and process `options.web|WebCredential`, even
  if those methods are changed by other software that runs after the
  polyfill is set up.
```

File: index.js [hunk 2]
```diff
@@ -88,6 +88,22 @@ export async function load(options = {
   } else {
     navigator.credentials = polyfill.credentials;
   }
+
+  // setup proxy for `navigator.credentials` to ensure subsequent changes to
+  // it do not prevent the polyfill from being able to run
```
Comment (TallTed) [inline index.js]:
```suggestion
  // set up proxy for `navigator.credentials`, to ensure subsequent changes
  // to it do not prevent the polyfill from running
```

**Review by TallTed — CHANGES_REQUESTED** [review]
Hopefully, clarifying

File: CHANGELOG.md (hunk 1 above)
Comment (dlongley) [reply CHANGELOG.md]: Yeah, I was thinking about fixing up that message on second read. Thanks for doing it for me, Ted!

**Review by jameseaster — APPROVED** [review]

File: index.js [hunk 3]
```diff
@@ -88,6 +88,26 @@ export async function load(options = {
...
+        obj[prop] = value;
+      }
+      // success
+      return true;
+    }
+  });
+
+  Object.defineProperty(
+    navigator, 'credentials', {value: navCredentialsProxy,writable: true}
+  );
```
Comment (dlongley) [inline index.js]: Style nits:

```suggestion
  Object.defineProperty(navigator, 'credentials', {
    value: navCredentialsProxy,
    writable: true
  });
```

File: index.js [hunk 4]
```diff
@@ -88,6 +88,26 @@ export async function load(options = {
...
     navigator.credentials = polyfill.credentials;
   }
+
+  // set up proxy for `navigator.credentials`, to ensure subsequent changes
+  // to it do not prevent the polyfill from running
+  const navCredentialsProxy = new Proxy(navigator.credentials, {
+    set(obj, prop, value) {
+      if(prop in polyfill.credentials._nativeCredentialsContainer) {
+        // replace underlying function, keeping credentials polyfill intact
+        polyfill.credentials._nativeCredentialsContainer[prop] = value;
```
Comment (dlongley) [inline index.js:98]: We might find that we need to use `defineProperty` here as well... but we can wait to hit that problem (if it exists out there), I guess.

**Review by jameseaster — APPROVED** [review]

**Pull request merged**